static/js/chart/draw_scatter.ts

```diff
@@ -550,6 +558,151 @@ function addRegressionLine(
   );
 }
 
+/**
+ * Adds text labels for a list of points
+ */
+function addTextLabels(
+  labelsGroup: d3.Selection<SVGGElement, any, any, any>,
+  xScale: ScatterScale,
+  yScale: ScatterScale,
+  points: Point[]
+): void {
+  labelsGroup
+    .attr("class", "dot-label")
+    .selectAll("text")
+    .data(points)
+    .enter()
+    .append("text")
+    .attr("dy", "0.35em")
+    .attr("x", (point) => xScale(point.xVal) + 7)
+    .attr("y", (point) => yScale(point.yVal))
+    .text((point) => point.place.name);
+}
+
+/**
+ * Gets at least MIN_HIGHLIGHT_POINTS points to highlight in each quadrant for a
+ * list of quadrants
+ */
+function getQuadrantHighlightPoints(
+  points: Point[],
+  xScale: ScatterScale,
+  yScale: ScatterScale,
+  quadrants: ChartQuadrant[]
+): Record<ChartQuadrant, Point[]> {
+  const numSectionsPerAxis = HIGHLIGHT_QUADRANT_SECTIONS * 2;
+  const xSectionLength =
+    (xScale.range()[1] - xScale.range()[0]) / numSectionsPerAxis;
+  const ySectionLength =
+    (yScale.range()[0] - yScale.range()[1]) / numSectionsPerAxis;
```

Comment:
nit: we should only call `.range()` once since it could get expensive with the number of points

Reply:
done